embdgen-core/src/embdgen/core/utils/class_factory.py

```diff
@@ -21,79 +23,86 @@ class Meta():
     optional: bool = False
 
     @classmethod
-    def get(cls, obj: Type) -> Dict[str, 'Meta']:
+    def get(cls, obj: Type) -> Dict[str, "Meta"]:
         return getattr(obj, cls._META_KEY, {})
 
     @classmethod
-    def set(cls, obj: Type, value: 'Meta'):
+    def set(cls, obj: Type, value: "Meta") -> None:
         name = value.name
         attr = cls.get(obj)
-        if cls._META_KEY not in obj.__dict__: # Test if the attribute is defined on the instance, not the parents
+        if cls._META_KEY not in obj.__dict__:  # Test if the attribute is defined on the instance, not the parents
             if hasattr(obj, cls._META_KEY):
-                attr = getattr(obj, cls._META_KEY).copy() # Use the parent's meta as preset
+                # Use the parent's meta as preset
+                attr = getattr(obj, cls._META_KEY).copy()
             setattr(obj, cls._META_KEY, attr)
         attr[name] = value
 
+
 def try_read_doc(cls: Type, name: str) -> str:
     try:
-        from sphinx.pycode import ModuleAnalyzer # pylint: disable=import-outside-toplevel
-    except ModuleNotFoundError: # pragma: no cover
+        from sphinx.pycode import ModuleAnalyzer  # pylint: disable=import-outside-toplevel
+    except ModuleNotFoundError:  # pragma: no cover
         return ""
     analyzer = ModuleAnalyzer.for_module(cls.__module__)
     analyzer.analyze()
     return "\n".join(analyzer.find_attr_docs().get((cls.__name__, name), [])).strip()
 
+
 CT = TypeVar("CT")
 
-def Config(name: str, doc="", optional=False):
+
+def Config(name: str, doc: str = "", optional=False) -> Callable[..., type[CT]]:
     """
     Metadata decorator for classes for the BaseFactory
     """
+
     def decorate(cls: Type[CT]) -> Type[CT]:
         local_doc = doc
-        name_prop = getattr(cls, name, None)
+        name_prop: Any | None = getattr(cls, name, None)
         if name_prop is not None and isinstance(name_prop, property):
             if name_prop.fset is None:
                 raise Exception(f"Property {name} in class {cls.__name__} has no setter defined")
-            setter_type = list(signature(name_prop.fset).parameters.values())[1].annotation
-            getter_type = signature(name_prop.fget).return_annotation
+            setter_type: Any = list(signature(name_prop.fset).parameters.values())[1].annotation
+            getter_type: Any = signature(obj=name_prop.fget).return_annotation
             if getter_type != setter_type:
                 raise Exception(f"Property {name} in class {cls.__name__} has different types for getter and setter")
             if not local_doc:
-                local_doc = name_prop.fget.__doc__ or ""
+                local_doc: str = name_prop.fget.__doc__ or ""
             hints = {name: getter_type}
         else:
-            hints = get_type_hints(cls)
+            hints: dict[str, Any] = get_type_hints(cls)
```

Comment:
Annotating here does not make sense. Any good editor derives the type from the function called. It does not make sense to annotate everything

Reply:
Actually it does, but seem you don't use MyPy. Currently it will end up with tons of errors over your code, unfortunately. So I don't even try to introduce it just yet, but doing small steps to have it.